stdlib/public/Concurrency/TaskStatus.cpp

```diff
@@ -525,6 +596,16 @@ static bool swift_task_hasTaskGroupStatusRecordImpl() {
 ///**************************************************************************/
 
 TaskExecutorRef AsyncTask::getPreferredTaskExecutor() {
+  // We first check the executor preference status flag, in order to avoid
+  // having to scan through the records of the task checking if there was
+  // such record.
+  //
+  // This is an optimization in order to make the enqueue/run
+  // path of a task avoid excessive work if a task had many records.
+  if (!hasTaskExecutorPreferenceRecord()) {
```

Comment:
This commit https://github.com/apple/swift/pull/68793/commits/4f16bdc8152e97eca12012278ab8f6882d486ba7 was the last thing I wanted to do here.

Since we query the task executor in functions like `defaultActorDrain` it would be a waste to for every task keep scanning all the records if most of them will not have a preference record.

We use 1 bit flag in the task status to indicate that there will be at least one preference record so we can avoid scanning the records all the time.

If this is a bad tradeoff (I thought making the "normal, no preferences" path be fast is preferable) I can revert this single commit. This is a piece I'd like some feedback on, thank you!